src/fiber/fiber.ml

```diff
@@ -426,8 +442,8 @@ module Ivar = struct
     | Full _ -> failwith "Fiber.Ivar.fill"
     | Empty q ->
       t.state <- Full x;
-      k ();
-      K.run_queue q x
+      EC.safe_run_k k ();
+      Queue.iter q ~f:(fun k -> K.run k x)
```

Comment:
How come we don't need to restore the execution context in the end anymore?

Reply:
Because this is always called in a tail position. No more code from the current fiber will be running after that.

Reply:
Does that break running `Fiber.run` more than once?

Probably not, because then then the tests would fail. But I'm not sure why not.

Reply:
The idea is that whenever we are about to restart some suspended continuation, we always restore the execution context. Also, `Fiber.run` uses a fresh execution context and always restore the previous one. So there is no risk of it leaking.